docs/specifications/tests/DNSSEC-TP/dnssec05.md

```diff
@@ -27,29 +31,25 @@ The DNSKEY record is defined in [RFC 4034, section 2].
 3. Repeat the following steps for each name server IP address:
 
    1. Send the DNSKEY query over UDP.
-   2. If no DNS response is returned, then emit *[NO_RESPONSE]*.
-   3. If the DNS response does not contain an DNSKEY RRset,
-      then emit *[NO_RESPONSE_DNSKEY]*.
-   4. Extract the algorithm numbers from each DNSKEY record and
-      compare the extracted algorithm number to the IANA
-      [DNSSEC Algorithm Numbers] registry.
-      1. If the algorithm is classified as "deprecated" (algorithm 
-         1), emit *[ALGORITHM_DEPRECATED]*.
-      2. If the algorithm is classified as "reserved" (algorithm 
-         4, 9, 11, 123-251 or 255), emit *[ALGORITHM_RESERVED]*.
-      3. If the algorithm is classified as "unassigned" (algorithm
-         17-122), emit *[ALGORITHM_UNASSIGNED]*.
-      4. If the algorithm is classified as "private algorithm"
-         (algorithm 253 or 254), emit *[ALGORITHM_PRIVATE]*.
-      5. If the algorithm is classified as "delete DS" (algorithm
-         0), emit *[ALGORITHM_DELETE_DS]*.
-      6. If the algorithm is classified as "indirect key" (algoritm
-         252), emit *[ALGORITHM_INDIRECT_KEY]*.
-      7. If the algorithm is not meant for zone signing (algorithm
-         0-2, 4, 9, 11, 17-252 or 255), emit 
-         *[ALGORITHM_NOT_ZONE_SIGN]*.
-      8. If no message has been emitted for the DNSKEY (algorithm
-         3, 5-8, 10, 12-16), emit *[ALGORITHM_OK]*.
+   2. If no DNS response is returned, then output *[NO_RESPONSE]*.
+   3. Else if the DNS response does not contain an DNSKEY RRset,
+      then output *[NO_RESPONSE_DNSKEY]*.
+   4. Else extract the algorithm numbers from each DNSKEY record and
+      compare the algorithm number to *Algorithm Status*.
+      1. If the algorithm is deprecated (algorithm 1, 3, 6 or 12)
+         output *[ALGORITHM_DEPRECATED]*.
+      2. If the algorithm is reserved (algorithm
+         4, 9, 11, 123-252 or 255), output *[ALGORITHM_RESERVED]*.
+      3. If the algorithm is unassigned (algorithm
+         17-122), output *[ALGORITHM_UNASSIGNED]*.
+      4. If the algorithm is private algorithm
+         (algorithm 253-254), output *[ALGORITHM_PRIVATE]*.
+      5. If the algorithm is not meant for zone signing (algorithm
+         0-2), output *[ALGORITHM_NOT_ZONE_SIGN]*.
+      6. If the algorithm is not rekommended for zone signing (algorithm
+         5, 7 or 10), output *[ALGORITHM_NOT_RECOMMENDED]*.
```

Comment:
Why did you remove INDIRECT_KEY?

Reply:
I did not remove it. It is treated as "reserved" instead. It is quite unclear what it really is. I guess it is better to move it back.